### PR TITLE
istioctl bug-report: Speed improvement from concurrency limiting

### PR DESCRIPTION
**Pull request by tonya11en**
The `istioctl bug-report` currently limits the request rate to the kube API to 10 RPS. This is rather conservative and causes the bug report generation to take an unreasonable amount of time in some cases. This patch speeds things up by limiting request **concurrency** instead of request **rate**. It results in dramatic speed increases in my local testing (1:12 -> 0:15).

To protect the api server from overload in cases where request concurrency is low and the rate is high, the bug-report will rely on server-side rate limiting to mitigate the overload risk.

Before:
```
istioctl bug-report  9.38s user 3.41s system 17% cpu 1:12.19 total
```

After:
```
out/linux_amd64/istioctl bug-report  10.54s user 3.70s system 97% cpu 14.566 total
```

### CODE REVIEW

**Comment by tonya11en** [comment]
I'll figure out this CLA thing tomorrow.

File: releasenotes/notes/bug-report-speedup.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# area describes the area that this change affects.
+# Valid values are:
+# - traffic-management
+# - security
+# - telemetry
+# - installation
+# - istioctl
+# - documentation
+area: telemetry
```
Comment (hanxiaop) [inline releasenotes/notes/bug-report-speedup.yaml:23]:
```suggestion
area: istioctl
```

File: tools/bug-report/pkg/bugreport/flags.go [hunk 2]
```diff
@@ -96,10 +96,9 @@ func addFlags(cmd *cobra.Command, args *config2.BugReportConfig) {
 	cmd.PersistentFlags().StringVar(&outputDir, "output-dir", "",
 		"Set a specific directory for output archive file.")
 
-	// requests per second limit
-	cmd.PersistentFlags().IntVar(&args.RequestsPerSecondLimit, "rps-limit", 0,
```
Comment (hanxiaop) [inline tools/bug-report/pkg/bugreport/flags.go:100]: I don't think we should remove flags directly like this without any deprecation notes. Can we provide an alternative option instead?

Comment (tonya11en) [reply tools/bug-report/pkg/bugreport/flags.go:100]: My reasoning for removing the rate is that it doesn't make much sense to follow any deprecation procedure the bug-report utility. Isn't this essentially a script that's run only in certain circumstances?

I'm happy to add it back if there are concerns around compatibility, though.

Comment (hanxiaop) [reply tools/bug-report/pkg/bugreport/flags.go:100]: Sounds good.

**Pull request merged**